functional/keylime_agent-UUID-providing/test.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+rlJournalStart
+
+    rlPhaseStartSetup "Do the keylime setup"
+        rlRun 'rlImport "./test-helpers"' || rlDie "cannot import keylime-tests/test-helpers library"
+        rlAssertRpm keylime
+        # update /etc/keylime.conf
+        limeBackupConfig
+        rlRun "limeUpdateConf logger_root level DEBUG"
+        rlRun "limeUpdateConf logger_keylime level DEBUG"
+        rlRun "limeUpdateConf handler_consoleHandler level DEBUG"
+
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+
+        # if TPM emulator is present
+        if limeTPMEmulated; then
+            # start tpm emulator
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            # make sure tpm2-abrmd is running
+            rlServiceStart tpm2-abrmd
+            sleep 5
+            # start ima emulator
+            rlRun "limeInstallIMAConfig"
+            rlRun "limeStartIMAEmulator"
+        fi
+        sleep 5
+        # start keylime_verifier
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier"
+        rlRun "limeStartRegistrar"
+        rlRun "limeWaitForRegistrar"
+    rlPhaseEnd
+    
+    rlPhaseStartTest "Hash_ek UUID"
+        AGENT_ID="hash_ek"
+        rlRun "limeUpdateConf agent  uuid '\"$AGENT_ID\"'"
+        rlRun "limeStartAgent"
```

Comment:
limeWaitForAgentRegistrationls move agent start before `limeWaitForAgentRegistration`

Reply:
`limeWaitForAgentRegistration` should be at end of the manual calculation of UUID, because we can verify by that way that providing of UUID by keylime is right. If the UUID was not assigned in the correct format or the manual calculation was incorrect, the agent would not register.

Reply:
Right. But we can start agent before it, i.e.
1. do ek hask calculations
2. configure agent
3. start agent
```
        AGENT_ID="hash_ek"
        #obtain ek cert and calculate has for assert
        rlRun "tpm2_nvread 0x1c00002 > ekcert.der"
        HASH_EK_UUID=$(openssl x509 -inform der -in ekcert.der -pubkey -noout | sha256sum | cut -d " " -f 1)
        rlLogInfo "HASH_EK_UUID=${HASH_EK_UUID}"
        # configure and start agent
        rlRun "limeUpdateConf agent  uuid '\"$AGENT_ID\"'"
        rlRun "limeStartAgent"
        rlRun "limeWaitForAgentRegistration ${HASH_EK_UUID}"
        rlRun "limeStopAgent"
```

Reply:
But is is not important, it just seemed to me being more "fluent" for reading.